docs/proposals/05-etcd-operator-tasks.md

```diff
@@ -0,0 +1,365 @@
+
+##### Task Config
+
+```go
+// SnapshotType can be full or delta snapshot
+SnapshotType string `json:"snapshotType"`
+const (
+  FullSnapshot SnapshotType = "full-snapshot"
+  DeltaSnapshot SnapshotType = "delta-snapshot"
+)
+
+type OnDemandSnapshotTaskConfig struct {
+  // Type of on-demand snapshot.
+  SnapshotType snapshotType `json:"snapshotType"`
+}
+```
+
+##### Pre-Conditions
+
+* Etcd cluster should have a quorum.
+* There should not already be a `on-demand snapshot` task running with the same `SnapshotType` for the same etcd cluster.
+
+#### Trigger on-demand maintenance of etcd cluster
+
+Operator can trigger on-demand maintenance of etcd cluster which includes operations like [etcd compaction](https://etcd.io/docs/v3.5/op-guide/maintenance/#history-compaction-v3-api-key-value-database), [etcd defragmentation](https://etcd.io/docs/v3.2/op-guide/maintenance/#defragmentation) etc.
+
+##### Possible Scenarios
+
+* If etcd cluster is heavily loaded which is causing performance degradation of etcd cluster, and operator doesn't want to wait for schedule maintenance window then operator can trigger on-demand maintenance of etcd cluster which internally calls the operations etcd-compaction, etcd-defragmentation etc. which will make etcd cluster lean and clean, therefore improves cluster performance.
+
+##### Task Config
+
+```go
+type OnDemandMaintenanceTaskConfig struct {
+  // MaintenanceType defines the maintenance operations need to be performed on etcd cluster.
+  MaintenanceType maintenanceOps
+}
+
+type maintenanceOps struct {
+  EtcdCompaction *bool `json:"etcd-compaction,omitempty"`
+  EtcdDefragmentation *bool `json:"defragmentation,omitempty"`
+}
+```
+
+##### Pre-Conditions
+
+* Etcd cluster should have a quorum.
+* There should not already be a duplicate task running with same `maintenanceType`.
+
+#### Copy the backups from one object store to another object store
+
+##### Task Config
+
+```go
+// EtcdCopyBackupsTaskConfig defines the parameters for the copy backups task.
+type EtcdCopyBackupsTaskConfig struct {
+  // SourceStore defines the specification of the source object store provider.
+  SourceStore StoreSpec `json:"sourceStore"`
+
+  // TargetStore defines the specification of the target object store provider for storing backups.
+  TargetStore StoreSpec `json:"targetStore"`
+
+  // MaxBackupAge is the maximum age in days that a backup must have in order to be copied.
+  // By default all backups will be copied.
+  // +optional
+  MaxBackupAge *uint32 `json:"maxBackupAge,omitempty"`
+
+  // MaxBackups is the maximum number of backups that will be copied starting with the most recent ones.
+  // +optional
+  MaxBackups *uint32 `json:"maxBackups,omitempty"`
+}
+```
+
+##### Pre-Conditions
+
+* There should not already be a duplicate task running.
+* It has no other pre-conditions as `copy-backup-task` runs as a seprate job, hence it doesn't depends on health of etcd cluster members or any other conditions.
```

Comment:
same as the above comment. Since this point is not a pre-condition it should be listed as one. See the above comment which has proposed changes.

Reply:
done